metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
@merrimanr  Whichever way we go, the interface has to change. I think we just need to decide the best way to do it.

I totally get the logic in adding `parseOptional` in that you don't want to break existing code.  But on the other hand, the project is young and we control all of the parsers. We can update and change them however we need to. 

When we just append methods to the interface (like `parseOptional`), we get interfaces that are not concise and clear.  To me that is a bigger problem that will only get worse with time.  We should bite the bullet now and make the interface change to something that is clear and concise.